DigitalLearningSolutions.Data.Tests/DataServices/GroupsDataServiceTests.cs

```diff
@@ -1,294 +1,344 @@
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                var removedDate = DateTime.Now;
+                const int delegateId = 299228;
+                AddDelegateToGroupWithSharedCourse();
+                AddProgressRecordForGroupWithSharedCourse();
+
+                // When
+                groupsDataService.RemoveRelatedProgressRecordsForGroupDelegate(8, delegateId, removedDate);
+                var progressFields = await connection.GetProgressRemovedFields(285172);
+
+                // Then
+                progressFields.Item1.Should().Be(0);
+                progressFields.Item2.Should().BeNull();
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [Test]
+        public void GetRelatedProgressIdForGroupDelegate_returns_expected_value()
+        {
+            // Given
+            const int delegateId = 245969;
+
+            // When
+            var result = groupsDataService.GetRelatedProgressIdForGroupDelegate(5, delegateId);
+
+            // Then
+            result.Should().Be(285146);
+        }
+
+        [Test]
+        public void GetRelatedProgressIdForGroupDelegate_returns_null_with_no_record()
+        {
+            // Given
+            const int delegateId = 2;
+
+            // When
+            var result = groupsDataService.GetRelatedProgressIdForGroupDelegate(1, delegateId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void UpdateGroupDescription_updates_record()
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                const int centerId = 101;
+                const int groupId = 5;
+                const string expectedDescription = "Test group description";
+
+                // When
+                groupsDataService.IsGroupDescriptionUpdated(groupId, centerId, expectedDescription);
+
+                // Then
+                var result = groupsDataService.GetGroupAtCentreById(groupId, centerId);
+                result?.GroupDescription.Should().Be(expectedDescription);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [Test]
+        public void UpdateGroupDescription_with_incorrect_centreId_does_not_update_record()
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                const int incorrectCentreId = 107;
+                const int groupId = 5;
+                const string expectedDescription = "Test group description";
```

Comment:
They used within try block and also they are consistent with other functions in the same class so would not be required to move outside try block

Reply:
Technically they aren't consistent because we have no consistency, but it's not really worth changing. I'm more inclined to remove as many of the thing that can't throw an error in the tests outside the try block assuming it doesn't leave us with split // Given groupings.